### PR TITLE
Base annotation

### PR DESCRIPTION
**Pull request by JennyVoPhamhi**
BaseAnnotation.py and demo_BaseAnnotation.ipynb are ready to review. (QuantiusAnnotation.py is also ready.) 

SpotAnnotationAnalysis.py is cleaned up and has very basic visualization functionality. (A lot of bugs had come into old visualization code due to changes to data structures.) demo_SpotAnnotationAnalysis.ipynb shows very basic visualization functionality. Declumping to be cleaned up and added.

In summary, still to come:
Visualization suites
Declumping methods

### CODE REVIEW

**Review by kevinyamauchi — CHANGES_REQUESTED** [review]
This is awesome! Some thoughts below.

- Please add a requirements.txt file
- For all functions, please look into using keyword arguments with default values (e.g., sa.plot_annotations).
- For all functions, please add type hints for the function signature
- Please move the declumping/cluster QC functions out of SpotAnnotation to stand alone functions
